src/careamics/lightning/dataset_ng/prediction/convert_prediction.py

```diff
@@ -120,7 +120,8 @@ def combine_samples(
 def convert_prediction(
     predictions: list[ImageRegionData],
     tiled: bool,
-) -> list[NDArray]:
+    return_sources: bool = False,
+) -> list[NDArray] | tuple[list[NDArray], list[str]]:
```

Comment:
I am personally not a fan of functions that return different structures depending on the arguments, but at least with overloads we shouldn't have to deal with checks on the union type elsewhere in the code

```suggestion
@overload
def convert_prediction(
    predictions: list[ImageRegionData],
    tiled: bool,
    return_sources: Literal[False],
) -> list[NDArray[Any]]: ...
@overload
def convert_prediction(
    predictions: list[ImageRegionData],
    tiled: bool,
    return_sources: Literal[True],
) -> tuple[list[NDArray[Any]], list[str]]: ...
def convert_prediction(
    predictions: list[ImageRegionData],
    tiled: bool,
    return_sources: bool = False,
) -> list[NDArray[Any]] | tuple[list[NDArray[Any]], list[str]]
```

Reply:
oops, first overload should also have the default value `return_sources: Literal[False] = False`

Reply:
I am liking the overload less and less, because I find them confusing and annoying to maintain. My brain only sees the same pattern and fails to register the difference. (Java requires a function body for overload, and that makes much more sense to me)

Best alternative in my mind: we just always return the sources and users can ignore them! We could return an empty list for arrays.

Reply:
Yeah I'm fine with always returning the sources. Anything that prevents having to do checks on the return type for mypy later on

Reply:
Ok, updated!